WebCryptoAPI/sign_verify/eddsa.js

```diff
@@ -1,363 +1,208 @@
-          }, "importVectorKeys step: " + vector.name + " verification");
-      });
-
-      all_promises.push(promise);
-  });
+      var algorithm = {name: vector.algorithmName};
 
-  // Test verification with an altered buffer after call
-  testVectors.forEach(function(vector) {
-      var promise = importVectorKeys(vector, ["verify"], ["sign"])
-      .then(function(vectors) {
-          var algorithm = {name: vector.algorithmName};
-          promise_test(function(test) {
+      // Test verification first, because signing tests rely on that working
+      promise_test(async() => {
+          let isVerified = false;
+          let key;
+          try {
+              key = await subtle.importKey("spki", vector.publicKeyBuffer, algorithm, false, ["verify"]);
+              isVerified = await subtle.verify(algorithm, key, vector.signature, vector.data)
+          } catch (err) {
+              assert_false(key === undefined, "importKey failed for " + vector.name + ". Message: ''" + err.message + "''");
+              assert_unreached("Verification should not throw error " + vector.name + ": " + err.message + "'");
+          };
+          assert_true(isVerified, "Signature verified");
+      }, vector.name + " verification");
+
+      // Test verification with an altered buffer after call
+      promise_test(async() => {
+          let isVerified = false;
+          let key;
+          try {
+              key = await subtle.importKey("spki", vector.publicKeyBuffer, algorithm, false, ["verify"]);
               var signature = copyBuffer(vector.signature);
-              var operation = subtle.verify(algorithm, vector.publicKey, signature, vector.data)
-              .then(function(is_verified) {
-                  assert_true(is_verified, "Signature verified");
-              }, function(err) {
-                  assert_unreached("Verification should not throw error " + vector.name + ": " + err.message + "'");
-              });
-
-              signature[0] = 255 - signature[0];
-              return operation;
-          }, vector.name + " verification with altered signature after call");
-      }, function(err) {
-          promise_test(function(test) {
-              assert_unreached("importVectorKeys failed for " + vector.name + ". Message: ''" + err.message + "''");
-          }, "importVectorKeys step: " + vector.name + " verification with altered signature after call");
-      });
-
-      all_promises.push(promise);
-  });
-
-  // Check for successful verification even if data is altered after call.
-  testVectors.forEach(function(vector) {
-      var promise = importVectorKeys(vector, ["verify"], ["sign"])
-      .then(function(vectors) {
-          var algorithm = {name: vector.algorithmName};
-          promise_test(function(test) {
+              isVerified = await subtle.verify(algorithm, key, signature, vector.data)
```

Comment:
```suggestion
              [isVerified] = await Promise.all([
                  subtle.verify(algorithm, key, signature, vector.data),
                  signature[0] = 255 - signature[0]
              ]);
```

Reply:
I'm not sure I have understood the point of altering the first signature's byte, since it's not checked any more in the test, is it ?